python/grass/script/testsuite/test_imagery.py

```diff
@@ -0,0 +1,144 @@
+from grass.exceptions import CalledModuleError
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+
+import grass.script as gs
+
+
+class TestImageryGroupToDict(TestCase):
+    """Tests function `group_to_dict` that returns raster maps
+    from an imagery group and their metadata."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.bands = [1, 2, 3]
+        cls.raster_maps = [f"lsat7_2002_{band}0" for band in cls.bands]
+        cls.group = "L8_group"
+        cls.subgroup = "L8_group_subgroup"
+        # Create input maps with label and group
+        for band in cls.bands:
+            cls.runModule(
+                "g.copy", raster=[f"lsat7_2002_{band}0", f"lsat7_2002_{band}0"]
+            )
+            cls.runModule(
+                "r.support", map=f"lsat7_2002_{band}0", semantic_label=f"L8_{band}"
+            )
+        cls.runModule("i.group", group=cls.group, input=cls.raster_maps)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.runModule("g.remove", type="raster", name=cls.raster_maps, flags="f")
+        cls.runModule("g.remove", type="group", name=cls.group, flags="f")
+
+    def test_basic_group_label_keys(self):
+        ref_dict = {f"L8_{band}": f"lsat7_2002_{band}0" for band in self.bands}
+        group_info = gs.imagery.group_to_dict(self.group, full_info=False)
+        # Check that a dict is returned
+        self.assertIsInstance(group_info, dict)
+        self.assertListEqual(list(ref_dict.keys()), list(group_info.keys()))
+        self.assertListEqual(
+            list(ref_dict.values()), [val.split("@")[0] for val in group_info.values()]
+        )
+
+    def test_invalid_input(self):
+        # Non existing group
+        self.assertRaises(
+            CalledModuleError, gs.imagery.group_to_dict, "non_existing_group"
+        )
+        # Invalid dict_keys
+        self.assertRaises(
+            ValueError,
+            gs.imagery.group_to_dict,
+            self.group,
+            **{"dict_keys": "invalid_dict_key"},
+        )
+        # Invalid dict_values
+        self.assertRaises(
+            ValueError,
+            gs.imagery.group_to_dict,
+            self.group,
+            **{"dict_values": "invalid_dict_value"},
+        )
+        group_info = gs.imagery.group_to_dict(
+            self.group, subgroup="non_existing_subgroup"
+        )
+
+        # Check that an empty dict is returned
+        self.assertIsInstance(group_info, dict)
+        self.assertDictEqual(group_info, {})
```

Comment:
My experience in other languages (not Python) is that these should be split as 5+ tests. It will really help pinpoint what is wrong when something fails (instead of 5+ possibilities). You already have it separated, and there isn't too much setup/parameters too, so I don't see why not.

Reply:
One style is called AAA: Arrange, Act, and Assert. So one setup, one action, and the assert (some purists may say a unit test should only have one assert, but I don't think it's always applicable).

Reply:
Thanks @echoix Good points. Now I did adjust the tests in [9e7923c](https://github.com/OSGeo/grass/pull/3756/commits/9e7923c9fc0213d5ca3259f5192643bdfaff469e).